rust/src/interpreter/utils.rs

```diff
@@ -0,0 +1,116 @@
+use std::cmp::min;
+
+use evmc_vm::{ExecutionMessage, MessageFlags, Revision, StatusCode};
+
+use crate::{interpreter::gas::consume_copy_cost, types::u256};
+
+pub trait SliceExt {
```

Comment:
nit: Please document and, if possible, suggest a better name for this. I would not know what a `SliceExt` is.

Reply:
Agreed. Will do that if a future PR.